stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -433,21 +434,53 @@ function versioninfo(io::IO=stdout)
     println(io, "LAPACK: ",Base.liblapack_name)
 end
 
-function __init__()
-    try
-        BLAS.check()
-        if BLAS.vendor() === :mkl
-            ccall((:MKL_Set_Interface_Layer, Base.libblas_name), Cvoid, (Cint,), USE_BLAS64 ? 1 : 0)
-        end
-        Threads.resize_nthreads!(Abuf)
-        Threads.resize_nthreads!(Bbuf)
-        Threads.resize_nthreads!(Cbuf)
-    catch ex
-        Base.showerror_nostdio(ex,
-            "WARNING: Error during initialization of module LinearAlgebra")
+function path_or_nothing(path, lib)
+    shlib_ext = string(".", Libdl.dlext)
+    fullpath = joinpath(path, string(lib, shlib_ext))
+    isfile(fullpath) ? fullpath : nothing
+end
+
+function get_blas_lapack_path()
+    blas_path = something(path_or_nothing(joinpath(Sys.BINDIR, Base.LIBDIR, "julia"),
+                                          Base.libblas_name),
+                          path_or_nothing(joinpath(Sys.BINDIR, Base.LIBDIR),
+                                          Base.libblas_name),
+                          path_or_nothing(Sys.BINDIR, Base.libblas_name),
+                          )
+
+    lapack_path = something(path_or_nothing(joinpath(Sys.BINDIR, Base.LIBDIR, "julia"),
+                                            Base.liblapack_name),
+                            path_or_nothing(joinpath(Sys.BINDIR, Base.LIBDIR),
+                                            Base.liblapack_name),
+                            path_or_nothing(Sys.BINDIR, Base.liblapack_name),
+                            )
```

Comment:
I suspect the problem is here: when `build_dir` is set to something different from `lib`, `joinpath(Sys.BINDIR, Base.LIBDIR, "julia")` doesn't exist before `make install`. So tests fail if they are run from the build dir. Not sure how to best handle this. Do we really need to compute the path? Usually I think we just rely on `dlopen` looking for the library in the right place.

If I create a link from `lib64` (which is `build_dir`) to `lib`, SuiteSparse tests pass.

Reply:
I tried with `dlpath` first and it actually crashes when called in init. So my plan was to compute paths until that gets fixed. I should be able to reproduce this issue, I think.

Reply:
If creating the symlink gets everything working, that would be great as a temporary measure. That doesn't feel too bad, given the magnitude of this change. But if I can do something here and try more paths, I am open to suggestions.

Reply:
@staticfloat Any suggestions on how best to address this?

Reply:
I wonder if @nalimilan can create the symlink necessary as part of the build. It might also break for other distro maintainers. Maybe we open an issue, and use `dlpath` once #39715 is fixed.

Reply:
Yeah I can work around the problem, and anyway I won't need this until 1.7 is released.

BTW, the whole `LIBBLASNAME` approach is actually broken since (last time I checked at least) things like Arpack.jl only work with keeping the default name. A solution would be to create a symlink called `libopenblas64_.so` (on 64-bit) or `libopenblas.so` (on 32-bit), pointing to the actual library specified by `LIBBLASNAME`.

Reply:
At least in BB, what we'll start doing is pointing everything to `libblastrampoline.so` whether it is 32-bit or 64-bit. So, hopefully this issue should just resolve itself once this PR and follow-on BB work happen. It should be possible to test it all out and ensure everything works before release.